src/app/reducers/totos.reducer.ts

```diff
@@ -0,0 +1,37 @@
+import { Action } from '@ngrx/store';
+
+import { Todo } from '../models/todo';
+import { TodosActions } from '../actions/todos.actions';
+
+export interface TodosState {
+  todos: Todo[];
+}
+
+const initialState: TodosState = {
+  todos: []
+};
+
+export function todos(state = initialState, action: Action): TodosState {
+  switch (action.type) {
+    case TodosActions.GET_TODOS_SUCCESS:
+      return Object.assign({}, state, {todos: action.payload});
+
+    case TodosActions.ADD_TODO_SUCCESS:
+      return Object.assign({}, state, {todos: state.todos.concat(action.payload)});
+
+    case TodosActions.TOGGLE_TODO_SUCCESS:
+
+      const newList = state.todos.map(todo => {
+        if (todo === action.payload) {
+          todo = new Todo(Object.assign({}, todo, {isDone: !todo.isDone}));
```

Comment:
you should NEVER reassign input parameters - it confuses maintainers and makes code harder to read